File: cs12b/lab2/FileReverse.java
//--------------------------------------------------------------------------------------------------------
// FileReverse.java
// Aryan Samuel
// arsamuel
// CMPS 12M
// 04/09/16
// Lab 2
// Takes input from a file, parses the tokens, and prints them out backwards each in a separate line.
//--------------------------------------------------------------------------------------------------------

import java.io.*;
import java.util.Scanner;

class FileReverse {
	
	public static void main (String[] args) throws IOException{
		
		Scanner in = null;
		PrintWriter out = null;
		String line = null;
		String[] token = null;
		int n;
		
		// Check CL argument length
		if (args.length < 2) {
			System.out.println("Usage: FileReverse <input file> <output file>");
			System.exit(1);
		}
		
		// Open files
		in = new Scanner(new File(args[0]));
		out = new PrintWriter(new FileWriter(args[1]));
		
		// Read lines from in and write to out
		while (in.hasNextLine()) {
			line = in.nextLine().trim() + " ";
			token = line.split("\\s+");
			for (int i=0; i<token.length; i++) {
				out.println(stringReverse(token[i], token[i].length()));
			}
		}
		
		// Close files
		in.close();
		out.close();
	}
	
	public static String stringReverse(String s, int n) {
		String S = "";
		if (n>0) {
			S = s.charAt(n-1)+stringReverse(s,n-1);
		} return S;
	} 
	
}

File: cs12b/lab3/FileReverse.c
/*
 * FileReverse.c
 * Aryan Samuel
 * arsamuel
 * CMPS 12M
 * 04/17/16
 * Lab 3
 * Takes string input from a user specifed file and reverse its contents.
 */
#include<stdio.h>
#include<stdlib.h>
#include<string.h>

/* Reverse string algorithm */
void stringReverse (char* s) {
	int i=0, j=strlen(s)-1;
	while (i<=j) {
		char temp = s[i];
		s[i] = s[j];
		s[j] = temp;
		i++;
		j--;
	}
}

/* Main function */
int main (int argc, char* argv[]) {
	FILE* in;
	FILE* out;
	char word[256];
	
	/* Command line argument check */
	if (argc != 3) {
	printf("Usage: %s <input file> <output file>\n", argv[0]);
	exit(EXIT_FAILURE);
	}
	
	/* Open all files */
	in = fopen(argv[1], "r");
	if (in==NULL) {
		printf("Unable to read from file %s\n", argv[1]);
		exit(EXIT_FAILURE);
	}
	
	out = fopen (argv[2], "w");
	if (out==NULL) {
		printf("Unable to write to file %s\n", argv[2]);
		exit(EXIT_FAILURE);
	}
	
	while (fscanf(in, "%s", word) != EOF) {
		stringReverse(word);
		fprintf(out, "%s\n", word);
	}
	
	/* Close all files */
	fclose(in);
	fclose(out);
	
	return(EXIT_SUCCESS);
}
